src/handlers/events-query.ts

```diff
@@ -0,0 +1,43 @@
+import type { DidResolver } from '../index.js';
+import type { EventLog } from '../types/event-log.js';
+import type { MethodHandler } from '../types/method-handler.js';
+import type { EventsQueryMessage, EventsQueryReply } from '../types/event-types.js';
+
+import { EventsQuery } from '../interfaces/events-query.js';
+import { messageReplyFromError } from '../core/message-reply.js';
+import { authenticate, authorize } from '../core/auth.js';
+
+
+export class EventsQueryHandler implements MethodHandler {
+
+  constructor(private didResolver: DidResolver, private eventLog: EventLog) { }
+
+  public async handle({
+    tenant,
+    message
+  }: {tenant: string, message: EventsQueryMessage}): Promise<EventsQueryReply> {
+    let eventsQuery: EventsQuery;
+
+    try {
+      eventsQuery = await EventsQuery.parse(message);
+    } catch (e) {
+      return messageReplyFromError(e, 400);
+    }
+
+    try {
+      await authenticate(message.authorization, this.didResolver);
+      await authorize(tenant, eventsQuery);
```

Comment:
Is selective sync only meant to allow one DID to sync data between their own DWNs? Or is it meant to allow two totally separate DWNs to sync e.g. a protocol context. If Alice has a protocol context, and Bob is trying to sync it, does he send an `EventsQuery` to Alice's DWN?

Reply:
Yes currently EventsQuery and EventsGet are scoped to only work for the DID's own DWN.